case-lib/lib.sh

```diff
@@ -366,18 +366,24 @@ func_lib_start_log_collect()
         # shellcheck disable=SC2034 # external script will use it
         SOF_LOG_COLLECT=1
     else
-        >&2 dlogw "without sudo permission to run $SOFLOGGER command"
+        >&2 dlogw "without sudo permission to run logging command"
         return 3
     fi
 
     if [ "X$is_etrace" == "X0" ];then
-        log_file=$LOG_ROOT/slogger.txt
-        log_opt="-t"
-        func_sof_logger_collect "$log_file" "$log_opt"
+        if is_ipc4 && is_firmware_file_zephyr; then
+            func_mtrace_collect
+        else
+            log_file=$LOG_ROOT/slogger.txt
+            log_opt="-t"
+            func_sof_logger_collect "$log_file" "$log_opt"
+        fi
     else
-        log_file=$LOG_ROOT/etrace.txt
-        log_opt=""
-        func_sof_logger_collect "$log_file" "$log_opt"
+        is_ipc4 || {
+            log_file=$LOG_ROOT/etrace.txt
+            log_opt=""
+            func_sof_logger_collect "$log_file" "$log_opt"
+        }
```

Comment:
I'm not following this syntact

Reply:
It means only non-ipc4 platforms have etrace support.